geniza/entities/admin.py

```diff
@@ -419,3 +478,66 @@ class PlacePlaceRelationTypeAdmin(TabbedTranslationAdmin, admin.ModelAdmin):
     fields = ("name",)
     search_fields = ("name",)
     ordering = ("name",)
+
+
+class EventDocumentInline(DocumentInline):
+    """Related documents inline for the Event admin"""
+
+    model = DocumentEventRelation
+    autocomplete_fields = ("document",)
+    fields = (
+        "document",
+        "document_description",
+        "notes",
+    )
+    extra = 0
+
+    def get_min_num(self, request, obj=None, **kwargs):
+        """Override min_num to change it conditionally based on how Event is being added"""
```

Comment:
I don't quite understand what this is doing or how it is used.  Does it only apply when creating a new event-document relationship?  (Your comments are almost enough but I can't quite get there)

Reply:
Updated comments to clarify. The idea here is that you can create an Event directly from a Document edit page, in an inline form, because all Events are supported by documentary evidence. For that reason, Events are also intended to always have at least one related Document. However, if we enforce that requirement at the DB level, or even in the admin using `min_num` out of the box, it would mean that you have to manually add the Event-Document relationship from the popup to save the Event without validation errors—even though you're creating it from within an inline on an associated Document's page, and therefore that relationship will be created as soon as you save the Document.

So it's more or less a convenience thing for content editors/admins who will be creating new Events directly from the popup on Document pages, so that they don't also have to choose that same Document in a dropdown within the popup to prevent validation errors.

The one caveat, which I've now noted, is that it's theoretically possible to create an Event from this inline popup and then _not_ save the Document, or delete the relationship with that new Event before saving the Document. In those cases, the Event would be orphaned, i.e. created without any relationship with a Document. But I figure that tradeoff is worth avoiding confusion ("why do I have to add a document to the Event when i'm creating the Event _from_ a document page?")